compiler/scenario-service/client/src/DA/Daml/LF/PrettyScenario.hs

```diff
@@ -46,6 +47,23 @@ type M = ExceptT Error (Reader (MS.Map NodeId Node, LF.World))
 
 type ModuleRef = LF.Qualified ()
 
+unmangleQualifiedName :: T.Text -> (LF.ModuleName, T.Text)
+unmangleQualifiedName t = case T.splitOn ":" t of
+    [modName, defName] -> (unmangleModuleName modName, unmangle defName)
+    _ -> error "Bad definition"
+
+unmangleModuleName :: T.Text -> LF.ModuleName
+unmangleModuleName t = LF.ModuleName (map unmangle $ T.splitOn "." t)
+
+unmangle :: T.Text -> T.Text
+unmangle s = case unmangleIdentifier s of
+    Left err -> error err
+    Right (UnmangledIdentifier s) -> s
+
+{-# COMPLETE UnmangledQualifiedName #-}
+pattern UnmangledQualifiedName :: LF.ModuleName -> T.Text -> TL.Text
+pattern UnmangledQualifiedName mod def <- (unmangleQualifiedName . TL.toStrict -> (mod, def))
```

Comment:
Throwing run time errors in patterns makes me a bit anxious...

Reply:
Yeah it’s not great but if this actually fails something is broken so I’d rather have it blow up than fall back into some other code path.